pkg/common/secret/utils.go

```diff
@@ -0,0 +1,51 @@
+package secret
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GenerateGenericSecret(secretName string, secretNamespace string,
+	key string, value string) *corev1.Secret {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: secretNamespace,
+		},
+		Data: map[string][]byte{
+			key: []byte(value),
+		},
+		Type: "Opaque",
+	}
+
+	return secret
+}
```

Comment:
I would prefer that we remove this function to avoid obfuscation and directly create the secret object where needed.

Reply:
But that would lead to a lot of duplicate code :/

Reply:
I am not too strong on this one, do what you feel is right. I expressed my concern.